src/NBitcoin/ChainedBlock.cs

```diff
@@ -62,11 +64,30 @@ public ChainedBlock(BlockHeader header, uint256 headerHash, ChainedBlock previou
             this.CalculateChainWork();
         }
 
+        /// <summary>
+        /// Constructs a chained block.
+        /// </summary>
+        /// <param name="header">The header for the chained block.</param>
+        /// <param name="height">The height of the chained block</param>
+        public ChainedBlock(BlockHeader header, int height)
+        {
+            this.Header = header ?? throw new ArgumentNullException("header");
+            this.Height = height;
+            this.HashBlock = header.GetHash();
+            this.CalculateChainWork();
```

Comment:
We might as well have the logic  of CalculateChainWork in the constructor. I've seen @dangershony request changes like this often ;)

Reply:
There are two constructors that call this though. I think I might be able to make one constructor call the other with a bit of work though. I will look into it and see if I can figure out something.

Reply:
i was thinking of that as well, but not sure how much work that is, if not too much, feel free to do it

Reply:
Think it's best to leave as two constructors, the logic is a bit different. So going to leave CalculateChainWork in method rather than duplicating the code in each constructor.

Reply:
sure no problem